File: app/src/main/java/com/originb/inkwisenote2/modules/admin/AdminUiState.java
package com.originb.inkwisenote2.modules.admin;

import java.io.File;
import java.util.List;

public abstract class AdminUiState {
    public static class DataList extends AdminUiState {
        public final List<?> data;
        public final String type;

        public DataList(List<?> data, String type) {
            this.data = data;
            this.type = type;
        }
    }

    public static class FilesState extends AdminUiState {
        public final File currentDir;
        public final List<File> files;

        public FilesState(File currentDir, List<File> files) {
            this.currentDir = currentDir;
            this.files = files;
        }
    }
}

File: app/src/main/java/com/originb/inkwisenote2/modules/admin/AdminUiState.kt
package com.originb.inkwisenote2.modules.admin

import java.io.File

abstract class AdminUiState {
    class DataList(val data: MutableList<*>?, val type: String?) : AdminUiState()

    class FilesState(val currentDir: File?, val files: MutableList<File?>?) : AdminUiState()
}
